src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,364 @@
+
+        elif (
+            self.measurement_class == MSMeasurement
+            or self.measurement_class == ReactorMeasurement
+            or self.measurement_class == Measurement
+        ):
+            obj_as_dict = self.read_ms()
+            if issubclass(cls, self.measurement_class):
+                self.measurement_class = cls
+            obj_as_dict.update(kwargs)
+            self.measurement = self.measurement_class.from_dict(obj_as_dict)
+
+            if self.mass_scans:
+                if self.verbose:
+                    print("adding mass scans to the measurement")
+                self.measurement = self.add_mass_scans()
+
+            # self.data_has_been_fetch = True
+
+            return self.measurement
+
+    def read_ms(self):
+        """Download MS data from cinfdata_database"""
+
+        self.cinf_db = plugins.cinfdata.connect(
+            setup_name=self.setup_name, grouping_column=self.grouping_column
+        )
+
+        self.group_data = self.cinf_db.get_data_group(
+            self.token, scaling_factors=(1e-3, None)
+        )
+
+        self.group_meta = self.cinf_db.get_metadata_group(self.token)
+        self.meta = self.group_meta[list(self.group_meta.keys())[0]]
+
+        self.set_sample_name()
+        self.set_name()
+        self.set_tstamp()
+
+        if self.verbose:
+            print("Retriving data from measurement named: ", self.sample_name)
+            print("Measurement started recording on: ", self.timestamp)
+
+        data_series_list = []
+        if self.verbose:
+            print("Column names in measurement: ")
+        for key in self.group_data.keys():
+            if self.group_meta[key]["type"] == 5:
+                column_name = self.group_meta[key]["mass_label"]
+                if self.verbose:
+                    print("Col name: ", column_name)
+                # unixtime = self.group_meta[key]["unixtime"]
+                # tstamp = float(unixtime)
+
+                tcol = self.group_data[key][:, 0]
+                vcol = self.group_data[key][:, 1]
+
+                tseries = TimeSeries(
+                    name=column_name + "-x",
+                    unit_name=get_column_unit(column_name + "-x") or "s",
+                    data=tcol,
+                    tstamp=self.tstamp,
+                )
+
+                vseries = ValueSeries(
+                    name=column_name,
+                    data=vcol,
+                    tseries=tseries,
+                    unit_name=get_column_unit(column_name + "-y"),
+                )
+                data_series_list.append(tseries)
+                data_series_list.append(vseries)
+            else:
+                pass
+
+        obj_as_dict = dict(
+            name=self.name,
+            sample_name=self.sample_name,
+            technique=self.technique,
+            reader=self,
+            series_list=data_series_list,
+            tstamp=self.tstamp,
+        )
+
+        if not obj_as_dict:
```

Comment:
ahh, yes. because self.technique has a value.

Would it be a better way to simply check whether data_series_list is empty to check if any data has been fetched?